server/routes/transferRouter.spec.js

```diff
@@ -0,0 +1,141 @@
+  afterEach(() => {
+    sinon.restore();
+  })
+
+  it("missing tokens should throw error", async () => {
+    const res = await request(app)
+      .post("/")
+      .send({
+        sender_wallet: "ssss",
+        receiver_wallet: "ssss",
+      });
+    expect(res).property("statusCode").eq(422);
+    expect(res.body.message).match(/bundle.*required/);
+  });
+
+  it("missing sender wallet should throw error", async () => {
+    const res = await request(app)
+      .post("/")
+      .send({
+        tokens: ["1"],
+        receiver_wallet: "ssss",
+      });
+    expect(res).property("statusCode").eq(422);
+    expect(res.body.message).match(/sender.*required/);
+  });
+
+  it("all parameters fine, should return 201", async () => {
+    sinon.stub(WalletService.prototype, "getByName").resolves(new Wallet(1));
+    sinon.stub(WalletService.prototype, "getById").resolves(new Wallet(2));
+    sinon.stub(TokenService.prototype,"getByUUID").resolves(new Token({
+      id: 1,
+      entity_id: 1,
+    }));
+    sinon.stub(Wallet.prototype, "transfer").resolves({
+      id: 1,
+      state: Transfer.STATE.completed,
+    });
+    sinon.stub(TransferService.prototype, "convertToResponse").resolves({
+      id:1,
+      state: Transfer.STATE.completed,
+    });
+    const res = await request(app)
+      .post("/")
+      .send({
+        tokens: ["1"],
+        sender_wallet: "ssss",
+        receiver_wallet: "ssss",
+      });
+    expect(res).property("statusCode").eq(201);
+  });
+
+  it("all parameters fine, but no trust relationship, should return 202", async () => {
+    sinon.stub(WalletService.prototype, "getByName").resolves(new Wallet(1));
+    sinon.stub(WalletService.prototype, "getById").resolves({
+      transfer: () => {},
+    });
+    sinon.stub(TokenService.prototype, "getByUUID").resolves(new Token({
+      id:1,
+      entity_id: 1,
+    }));
+    WalletService.prototype.getById.restore();    
+    sinon.stub(WalletService.prototype, "getById").resolves({
+      transfer: async () => {
+        throw new HttpError(202);
+      },
+    });
+    const res = await request(app)
+      .post("/")
+      .send({
+        tokens: ["1"],
+        sender_wallet: "ssss",
+        receiver_wallet: "ssss",
+      });
+    expect(res).property("statusCode").eq(202);
+  });
+
+  it("bundle case, success, should return 201", async () => {
+    sinon.stub(WalletService.prototype, "getByName").resolves(new Wallet(1));
+    sinon.stub(WalletService.prototype, "getById").resolves({
+      transfer: () => {},
+    });
+    sinon.stub(TokenService.prototype, "getByUUID").resolves(new Token({
+      id:1,
+      entity_id: 1,
+    }));
+    WalletService.prototype.getById.restore();    
+    sinon.stub(WalletService.prototype, "getById").resolves({
+      transferBundle: async () => {
+        throw new HttpError(202);
+      },
+    });
+    const res = await request(app)
+      .post("/")
+      .send({
+        bundle: {
+          bundle_size: 1,
```

Comment:
Should we test what will happen with bundle size > 1 and what gets transferred?

Reply:
Yes! that makes sense.